### ISSUE TITLE
Uncaught DOMException: Failed to execute 'querySelectorAll' on 'Document': '#' is not a valid selector.

### ISSUE THREAD

**Issue by dsincl12** [phoenixframework/phoenix#5604]
### Environment

* Elixir version (elixir -v): Elixir 1.15.7 (compiled with Erlang/OTP 26)
* Phoenix version (mix deps): 1.7.9
* Operating system: macOS Sonoma 14.0

### Actual behavior

After creating a new Phoenix app with mix phx.new and adding authentication with mix phx.gen.auth, I noticed that I was unable to close the flash message after I was logged in. Looking at the console in Chrome I see the following:

```
Uncaught DOMException: Failed to execute 'querySelectorAll' on 'Document': '#' is not a valid selector.
    at Object.all (http://localhost:4000/assets/app.js:1844:35)
    at Object.filterToEls (http://localhost:4000/assets/app.js:4126:31)
    at http://localhost:4000/assets/app.js:3924:14
    at Array.forEach (<anonymous>)
    at Object.exec (http://localhost:4000/assets/app.js:3919:16)
    at http://localhost:4000/assets/app.js:5764:24
    at http://localhost:4000/assets/app.js:5548:37
    at LiveSocket.owner (http://localhost:4000/assets/app.js:5544:9)
    at LiveSocket.withinOwners (http://localhost:4000/assets/app.js:5548:12)
    at http://localhost:4000/assets/app.js:5763:16
```

when clicking on the X to close the flash.

### Expected behavior

The flash to close without any error messages in the console.

**Comment by dinocosta** [comment]
@dsincl12 I haven't reproduced this issue myself but looks like https://github.com/phoenixframework/phoenix/pull/5602 might fix it? It does indeed look like it's trying to select an element using it's id, even though there's no id, so that's probably why there's nothing after `#` in the output - `'#' is not a valid selector.` 🤔

**Comment by gustavomtborges** [comment]
> @dsincl12 I haven't reproduced this issue myself but looks like #5602 might fix it? It does indeed look like it's trying to select an element using it's id, even though there's no id, so that's probably why there's nothing after `#` in the output - `'#' is not a valid selector.` 🤔

The #5602 fixed for me!

**Comment by Gazler** [comment]
I've merged #5602 - thanks for checking it.

**Comment by orikremer** [comment]
Confirming this solved the issue for me.
